### ISSUE TITLE
Handle integration failure

### ISSUE THREAD

**Issue by Silmathoron** [nest/nestml#485]
Using Python 3.6, the code generated by NESTML for GSL does not seem to handle the generation of infinities or NaN values inside the GSL solver.
In these cases, it seems that GSL_SUCCESS is still returned but the time is not incremented, leading to an infinite loop.
Because NESTML will probably mostly be used for model development (which may often lead to such issues), it would be useful to handle this case (possibly only if the ``dev`` keyword is passed).

I don't know how to do this from the GSL side, a workaround would be the addition of a counter condition for the while loop (which would be added only if ``dev=True`` is passed).

**Comment by clinssen** [comment]
Checking for NaNs and Infs should probably occur at the end of the ODE stepping function (`[model_name]_dynamics()`). Two checks are required for each element of the state vector (one for Inf, one for NaN). Proposal: these checks should be present by default but can be suppressed by supplying an `--optimize` command-line parameter.

**Comment by Silmathoron** [comment]
@clinssen OK, these tests sounds better than my original proposal, I'm a bit unsure about including them by default, though... plus the nice thing about the ugly counter workaround is that it lets the simulation run through and allows the user to see more precisely where it went wrong...

**Comment by clinssen** [comment]
See also a similar but slightly different strategy employed in `aeif_cond_alpha`: https://github.com/nest/nest-simulator/blob/master/models/aeif_cond_alpha.cpp#L516

**Comment by Silmathoron** [comment]
Yeah, I must say I'm not really convinced by this, I would rather have the tests you proposed :+1:
